cognite_toolkit/_cdf_tk/commands/build_cmd.py

```diff
@@ -663,6 +673,13 @@ def check_built_resource(
 
         return warning_list, identifier_kind_pairs
 
+    def warn(self, warning: ToolkitWarning, include_timestamp: bool = False, console: Console | None = None) -> None:
+        self.warning_list.append(warning)
+        if self.print_warning or self.exit_on_warning:
+            warning.print_warning(include_timestamp, console)
+            if self.exit_on_warning:
+                raise typer.Exit(code=1)
+
```

Comment:
This exists on the first possible warning, which is not what we want. Instead check when build is over if there are any warnings and exit on then.